OrbitQt/orbitglwidget.cpp

```diff
@@ -34,6 +34,12 @@ bool OrbitGLWidget::eventFilter(QObject* /*object*/, QEvent* event) {
       }
     }
   }
+
+  if (gl_canvas_ && gl_canvas_->GetNeedsCheckHighlightChange()) {
+    checkFunctionHighlightChange();
+    gl_canvas_->ResetNeedsCheckHighlightChange();
+  }
```

Comment:
the check here is performed on every event, it should probably be done only on the paint event. Actually, I find this is a bit convoluted, could we simplify by keeping track of the canvas' selected function address, and running the highlight logic when we detect it has changed?

Reply:
Totally agree and thanks for pointing out this. I had a meant to do this check for every mouse events that might incur the selection of a textbox, but I didn't find a suitable if condition. Thanks for explaining this to me.